x-pack/solutions/security/plugins/security_solution/common/siem_migrations/model/api/rules/rule_migration.schema.yaml

```diff
@@ -99,34 +84,93 @@ paths:
+            application/json:
+              schema:
+                $ref: '../../rule_migration.schema.yaml#/components/schemas/RuleMigration'
+        404:
+          description: Indicates the migration id was not found.
+    get:
+      summary: Retrieves a rule migration
+      operationId: GetRuleMigration
+      x-codegen-enabled: true
+      x-internal: true
+      description: Retrieves the rule migration document stored in the system given the rule migration id
+      tags:
+        - SIEM Rule Migrations
+      parameters:
+        - name: migration_id
+          in: path
+          required: true
+          schema:
+            description: The migration id to retrieve
+            $ref: '../../../../../common/api/model/primitives.schema.yaml#/components/schemas/NonEmptyString'
+      responses:
+        200:
+          description: Indicates rules migrations have been retrieved correctly.
+          content:
+            application/json:
+              schema:
+                $ref: '../../rule_migration.schema.yaml#/components/schemas/RuleMigration'
+        404:
+          description: Indicates the migration id was not found.
+    delete:
+      summary: Deletes a rule migration and its associated resources
+      operationId: DeleteRuleMigration
+      x-codegen-enabled: true
+      x-internal: true
+      description: Deletes a rule migration document stored in the system given the rule migration id
+      tags:
+        - SIEM Rule Migrations
+      parameters:
+        - name: migration_id
+          in: path
+          required: true
+          schema:
+            description: The migration id to delete
+            $ref: '../../../../../common/api/model/primitives.schema.yaml#/components/schemas/NonEmptyString'
+      responses:
+        200:
+          description: Indicates rules migrations have been deleted correctly.
+        404:
+          description: Indicates the migration id was not found.
+
+  /internal/siem_migrations/rules/{migration_id}/rules:
+    post:
+      summary: Add rules to a rule migration
+      operationId: CreateRuleMigrationRules
+      x-codegen-enabled: true
+      x-internal: true
+      description: Adds original vendor rules to an already existing migration. Can be called multiple times to add more rules
+      tags:
+        - SIEM Rule Migrations
+      parameters:
+        - name: migration_id
+          in: path
+          required: true
+          schema:
+            description: The migration id to create rules for
+            $ref: '../../../../../common/api/model/primitives.schema.yaml#/components/schemas/NonEmptyString'
       requestBody:
         required: true
         content:
           application/json:
             schema:
               type: array
               items:
-                $ref: '../../rule_migration.schema.yaml#/components/schemas/UpdateRuleMigrationData'
+                $ref: '../../rule_migration.schema.yaml#/components/schemas/OriginalRule'
```

Comment:
Shouldn't this be
```suggestion
                $ref: '../../rule_migration.schema.yaml#/components/schemas/UpdateRuleMigrationRule'
```

Reply:
or am I getting confused with the diff? 😕

Reply:
Ya, diff is indeed confusing. But it should be `$ref: '../../rule_migration.schema.yaml#/components/schemas/OriginalRule'` because this endpoint is for creating the rules, hence array of original Rules should be the input.

Line 269 is actually where `UpdateRuleMigrationRules` endpoint starts.